### ISSUE TITLE
Undefined -var variable building ubuntu22.04 image

### ISSUE THREAD

**Issue by cengelenVW** [YannickRe/azuredevops-buildagents#60]
Hi,
Last month the team at Microsoft has apparently been busy changing their *.pkr.hcl files, and those changes are causing issues using the pipelines in this repository.
Today I've tried to generate a released ubuntu22.04 image, but it fails at the beginning of the `Build VM` step:
```
Syntax-only check passed. Everything looks okay.
Show Packer Version
1.9.2
Build ubuntu2204 VM
Error: Undefined -var variable

A "storage_account" variable was passed in the command line but was not found in
known variables. To declare variable "storage_account", place this block in one
of your .pkr files, such as variables.pkr.hcl

Error: Undefined -var variable

A "resource_group" variable was passed in the command line but was not found in
known variables. To declare variable "resource_group", place this block in one
of your .pkr files, such as variables.pkr.hcl

Error: Undefined -var variable

A "capture_name_prefix" variable was passed in the command line but was not
found in known variables. To declare variable "capture_name_prefix", place this
block in one of your .pkr files, such as variables.pkr.hcl
```

Last month I've successfully generated an image. So the conflicting change must be made between those tags. A diff between those tags: https://github.com/actions/runner-images/compare/ubuntu22/20230806.1...ubuntu22/20230903.1?diff=split#diff-d91b5305d8b642adb5996f611a68484ffe1354d76246d50fc0395e8d3c5ec669

I think the build-image.ps1 needs an update. Hopefully someone is able to fix this.

**Comment by YannickRe** [comment]
It's probably a specific result of what @b-twis attended me about in #59 .
They have postponed this transition to the last possible minute, which doesn't really work well with my current commitments. My goal is to get this repository working again ASAP in the new way, we depend on it ourselves. I just might not make the deadline.
Hopefully y'all can keep working a while with your existing images 🙁

**Comment by jvanegmond** [comment]
For anyone bumping into this issue and looking to build the last known working version, you can replace `git checkout ($latest_release.tag_name)` in `buildagent-generation-template.yaml#93` with the specific release that you want to use. For example for ubuntu22.04 there is `ubuntu22/20230821.1`. Keep in mind that this value changes depending on `$os_image_type` and `$runner_images_version` and that this workaround does mean you will be behind on agent updates until issue #59 and #60 are fixed.

**Comment by erik-de-bont** [comment]
I have update the code so that we should be able to generate runner images again. See https://github.com/YannickRe/azuredevops-buildagents/pull/62

**Comment by YannickRe** [comment]
It has been merged in #63
